File: app/src/main/java/com/fastaccess/provider/timeline/handler/QuoteHandler.java
package com.fastaccess.provider.timeline.handler;

import androidx.annotation.ColorInt;

import android.text.SpannableStringBuilder;

import com.zzhoujay.markdown.style.MarkDownQuoteSpan;

import net.nightwhistler.htmlspanner.TagNodeHandler;

import org.htmlcleaner.TagNode;

import lombok.AllArgsConstructor;

/**
 * Created by Kosh on 23 Apr 2017, 11:30 AM
 */

@AllArgsConstructor public class QuoteHandler extends TagNodeHandler {

    @ColorInt private int color;

    @Override
    public void handleTagNode(TagNode node, SpannableStringBuilder builder, int start, int end) {
        try {
            builder.setSpan(new MarkDownQuoteSpan(color), start + 1, builder.length(), 33);
        } catch (IndexOutOfBoundsException e) {
            builder.setSpan(new MarkDownQuoteSpan(color), start, builder.length(), 33);
        }
    }
}

File: markdown/src/main/java/com/fastaccess/markdown/spans/QuoteHandler.kt
package com.fastaccess.markdown.spans

import android.text.SpannableStringBuilder
import net.nightwhistler.htmlspanner.SpanStack

import net.nightwhistler.htmlspanner.TagNodeHandler

import org.htmlcleaner.TagNode

/**
 * Created by Kosh on 23 Apr 2017, 11:30 AM
 */

class QuoteHandler(private val color: Int = 0) : TagNodeHandler() {
    override fun handleTagNode(
        node: TagNode?,
        builder: SpannableStringBuilder?,
        start: Int,
        end: Int,
        spanStack: SpanStack?
    ) {
        builder?.setSpan(MdQouteSpan(color), if (start > builder.length - 1) start + 1 else start, builder.length - 1, 33)
    }
}
